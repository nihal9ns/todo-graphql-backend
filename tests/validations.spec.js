const mongoose = require("mongoose");
const ToDoModel = require("../models/ToDo");
const { checkIfToDoExists } = require("../validations");
const { todos } = require("./mocks");

describe("TEST validations", () => {
  beforeAll(async () => {
    const url = "mongodb://localhost/test-todo-app";

    await mongoose.connect(url, {
      useNewUrlParser: true,
      useUnifiedTopology: true
    });

    await ToDoModel.insertMany(todos);
  });

  afterAll(async () => {
    await ToDoModel.deleteMany({});
    await mongoose.connection.close();
  });

  it("Should return true if todo exist", async () => {
    const todo = await ToDoModel.find({});
    const foundToDo = await checkIfToDoExists(todo[0]._id);
    expect(foundToDo).toBeDefined();
    expect(foundToDo).toBeTruthy();
  });

  it("Should throw error if todo does not exist", async () => {
    try {
      const id = "5d99919960dd0d698dbe228c"; // does not exist
      await checkIfToDoExists(id);
    } catch (err) {
      expect(err).toMatchSnapshot();
    }
  });
});
