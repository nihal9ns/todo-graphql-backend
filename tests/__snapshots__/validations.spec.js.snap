// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST validations Should throw error if todo does not exist 1`] = `[Error: ToDo with id 5d99919960dd0d698dbe228c does not exist!]`;
